### PR TITLE
Add support for adding spotlight label to beatmaps

### PR DESCRIPTION
**Pull request by venix12**
green one :eyes: (maybe pink would make more sense to match the spotlights marker in lazer?)

- needs to queue reindex when setting spotlight value, should there be an on-site UI for that perhaps?
- should run ``php artisan es:index-documents --types=beatmapsets --yes`` after merging

part of [#8736](https://github.com/ppy/osu-web/issues/8736)

### CODE REVIEW

**Comment by peppy** [comment]
I think this should be fine to get merged in to see how it works?

@venix12 One question I have is whether the intention here is to mark any beatmap that was ever included in a spotlight, or whether to only mark active spotlight beatmaps?

**Comment by Ephemeralis** [comment]
Marking any map ever featured in a spotlight has been the intention historically.

**Comment by venix12** [comment]
could also have them be a different colour or so for the active ones in the future (when the seasons system is in place, would just take playlist items of ongoing ones)

File: database/migrations/2022_04_04_152502_add_spotlight_to_osu_beatmapsets.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
```
Comment (nanaya) [inline database/migrations/2022_04_04_152502_add_spotlight_to_osu_beatmapsets.php:1]: update the date in filename

also the `osu_`/`Osu` prefix can be removed

(optionally change the class into anonymous class like in `2022_04_12_...` migration)

File: resources/assets/lib/components/beatmapset-panel.tsx [hunk 2]
```diff
@@ -67,9 +67,9 @@ const BeatmapDots = observer(({ compact, beatmaps, mode }: { beatmaps: BeatmapJs
   </div>
 ));
 
-const FeaturedArtistBadge = () => (
-  <span className='beatmapset-badge beatmapset-badge--featured-artist beatmapset-badge--panel'>
-    {osu.trans('beatmapsets.featured_artist_badge.label')}
+const BeatmapsetBadge = ({ type }: { type: string }) => (
+  <span className={`beatmapset-badge beatmapset-badge--${type} beatmapset-badge--panel`}>
+    {osu.trans(`beatmapsets.${type.replace('-', '_')}_badge.label`)}
```
Comment (nanaya) [inline resources/assets/lib/components/beatmapset-panel.tsx:72]: `replaceAll` instead of `replace`

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — CHANGES_REQUESTED** [review]

**Comment by nanaya** [comment]
Conflicts 👀

**Review by nanaya — APPROVED** [review]
this involves adding a boolean column `spotlight` to `osu_beatmapsets`. It takes a bit on my system so if to be done manually, the migration entry will be `2022_04_30_152502_add_spotlight_to_beatmapsets`

**Review by peppy — APPROVED** [review]

**Pull request merged**